app/assets/javascripts/reducers/users.js

```diff
@@ -23,12 +23,42 @@ const SORT_DESCENDING = {
 
 export default function users(state = initialState, action) {
   switch (action.type) {
-    case RECEIVE_USERS: return {
+    case RECEIVE_USERS: {
+      // Get the sorting key if available from Redux store or else use last_name
```

Comment:
The sorting logic for this is still quite hard to understand. That's mostly because of how we don't actually store sorting direction in state, we just compare the new sorting from the SORT_USERS action with the not-yet-updated state.

I think this needs to be refactored to make it easier to follow, and to be more explicit about whether to do reverse sorting.

I suggest:
* handling `user_list` first.
* next, determine the sort key (`sort.key` or `last_name` or `username`).
* next, determine whether the previous sorting was a reversed sort or not, via an explicitly named boolean. (It was reversed if `sort.key` is different from `sort.sortKey`, I think.
* finally, do the sortByKey operation and use the `isReversed` (or whatever name) boolean to decide whether to pass the sort key or `null` as the `previousKey` argument. It will do reverse sorting if the same value is passed for both sortKey and previousKey.

Reply:
Okay